app/models/concerns/rdf/graphable.rb

```diff
@@ -0,0 +1,169 @@
+      #
+      # @example
+      #   class EDM::ProvidedCHO
+      #     include Mongoid::Document
+      #     field :dc_subject_agents, class_name: 'EDM::Agent'
+      #     has_rdf_predicate(:dc_subject_agents, RDF::Vocab::DC11.subject)
+      #   end
+      def has_rdf_predicate(field_name, rdf_predicate)
+        rdf_predicates[field_name] = rdf_predicate
+      end
+
+      def rdf_predicates
+        @rdf_predicates ||= HashWithIndifferentAccess.new
+      end
+
+      def rdf_type
+        @rdf_type ||= begin
+          vocab = RDF::Vocab.const_get(to_s.deconstantize)
+          vocab.send(to_s.demodulize)
+        end
+      end
+
+      def rdf_fields_and_predicates
+        @rdf_fields_and_predicates ||= begin
+          fields_and_relations.keys.each_with_object(HashWithIndifferentAccess.new) do |field_name, memo|
+            memo[field_name] = rdf_predicate_for_field(field_name)
+          end.reject do |_field_name, rdf_predicate|
+            rdf_predicate.nil?
+          end
+        end
+      end
+
+      def rdf_predicate_for_field(field_name)
+        return rdf_predicates[field_name] if rdf_predicates.key?(field_name)
+
+        PREFIXED_VOCABULARIES.each do |prefix, vocab|
+          match = field_name.to_s.match(/\A#{prefix}_(.+)\z/)
+          next if match.nil?
+          return vocab.respond_to?(match[1]) ? vocab.send(match[1]) : nil
+        end
+        nil
+      end
+
+      def fields_and_relations
+        @fields_and_relations ||= fields.merge(relations)
+      end
+    end
+
+    included do
+      delegate :rdf_fields_and_predicates, :fields_and_relations, to: :class
+    end
+
+    def to_rdf
+      rdf_graph = RDF::Graph.new.tap do |graph|
+        graph << [rdf_uri, RDF.type, self.class.rdf_type]
+        rdf_fields_and_predicates.each_pair do |field_name, predicate|
+          next if exclude_from_rdf_output?(predicate)
+          field = fields_and_relations[field_name]
+          field_graph = rdf_graph_for_field(field)
+          graph.insert(field_graph) unless field_graph.nil?
+        end
+      end
+
+      literalized_rdf_graph(rdf_graph) || rdf_graph
+    end
+
+    def rdf_graph_for_field(field)
+      rdf_predicate = rdf_fields_and_predicates[field.name.to_s]
+
+      if field.is_a?(Mongoid::Fields::Localized)
+        rdf_graph_for_localized_field(field, rdf_predicate)
+      else
+        rdf_graph_for_unlocalized_field(field, rdf_predicate)
+      end
+    end
+
+    def rdf_graph_for_localized_field(field, rdf_predicate)
+      field_value = send("#{field.name}_translations")
+      return if field_value.blank?
+
+      RDF::Graph.new.tap do |graph|
+        field_value.each_pair do |language, value|
+          graph << [rdf_uri, rdf_predicate, rdf_uri_or_literal(value, language: language)]
+        end
+      end
+    end
+
+    def rdf_graph_for_unlocalized_field(field, rdf_predicate)
+      field_value = send(field.name)
+      return if field_value.nil? || field_value == ''
```

Comment:
could we use '.blank?' or is that too restrictive for some values like ' ', or empty arrays?

Reply:
`#blank?` should be fine.